contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -406,4 +454,59 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
   function isCapabilityDeprecated(bytes32 hashedCapabilityId) external view returns (bool) {
     return s_deprecatedHashedCapabilityIds.contains(hashedCapabilityId);
   }
+
+  /// @notice Adds a DON made up by a group of nodes that support a list
+  /// of capability configurations
+  /// @param nodes The nodes making up the DON
+  /// @param capabilityConfigurations The list of configurations for the
+  /// capabilities supported by the DON
+  /// @param isPublic True if the DON is public
+  function addDON(
+    bytes32[] calldata nodes,
+    CapabilityConfiguration[] calldata capabilityConfigurations,
+    bool isPublic
+  ) external onlyOwner {
+    uint32 id = s_donId;
+
+    s_dons[id].id = id;
+    s_dons[id].isPublic = isPublic;
+
+    for (uint256 i; i < capabilityConfigurations.length; ++i) {
+      CapabilityConfiguration calldata configuration = capabilityConfigurations[i];
+      bytes32 capabilityIds = configuration.capabilityId;
+
+      if (s_capabilities[capabilityIds].version == bytes32("")) revert CapabilityDoesNotExist(capabilityIds);
+
+      if (s_dons[id].capabilityIds.contains(capabilityIds)) revert DonCapabilityExists(id, capabilityIds);
+
+      s_dons[id].capabilityIds.add(capabilityIds);
+      s_dons[id].capabilityConfigs[capabilityIds] = configuration.config;
+    }
+
```

Comment:
TODO: Verify capability is not deprecated https://chainlink-core.slack.com/archives/C06F7S6T2BU/p1714585501653989